lib/optimizely/odp/odp_segments_api_manager.rb

```diff
@@ -41,8 +41,13 @@ def fetch_segments(api_key, api_host, user_key, user_value, segments_to_check)
       headers = {'Content-Type' => 'application/json', 'x-api-key' => api_key.to_s}
 
       payload = {
-        'query' => %'query {customer(#{user_key}: "#{user_value}")' \
-                   "{audiences(subset:#{segments_to_check || '[]'}) {edges {node {name state}}}}}"
+        query: 'query($vuid: String, $fs_user_id: String, $audiences: [String]) {' \
+               'customer(vuid: $vuid, fs_user_id: $fs_user_id) ' \
+               '{audiences(subset: $audiences) {edges {node {name state}}}}}',
+        variables: {
+          user_key => user_value.to_s,
+          audiences: segments_to_check || []
```

Comment:
Can you take a look at the format in csharp SDK? https://github.com/optimizely/csharp-sdk/blob/8f23e031c5b766ea8ba678d55114be3cf9c3a494/OptimizelySDK/Odp/OdpSegmentApiManager.cs#L152
This one is more clean in that we do not list all accepted query-ids. Let me know if you see any concern. It'll be good if we can use the same format for all SDKs.

Reply:
Yea I think that is the better solution, I'll update this one to match.